src/Stack/Dot.hs

```diff
@@ -140,28 +140,37 @@ listDependencies opts = do
   if listDepsTree opts then
     do
       liftIO $ Text.putStrLn "Packages"
-      liftIO $ printTree opts 0 [] pkgs resultGraph
+      liftIO $ printTree opts 0 [] (treeRoots opts pkgs) resultGraph
     else
       void (Map.traverseWithKey go (snd <$> resultGraph))
       where go name payload = liftIO $ Text.putStrLn $ listDepsLine opts name payload
 
+treeRoots :: ListDepsOpts -> Set PackageName -> Set PackageName
+treeRoots opts defaults = let targets = dotTargets $ listDepsDotOpts opts
+                     in if null targets
+                           then defaults
+                           else Set.fromList $ map (mkPackageName . Text.unpack) targets
+
 printTree :: ListDepsOpts
           -> Int
           -> [Int]
           -> Set PackageName
           -> Map PackageName (Set PackageName, DotPayload)
           -> IO ()
-printTree opts depth remainingDepsCounts packages dependencyMap =
+printTree opts depth remainingDepsCounts packages dependencyMap = do
```

Comment:
This is unnecessary — you only have a single expression.

```suggestion
printTree opts depth remainingDepsCounts packages dependencyMap =
```

Reply:
Also I'm going to have to take a break until tomorrow; I think this function could be clearer.